hw/top_earlgrey/ip/ast/doc/_index.md

```diff
@@ -0,0 +1,1006 @@
+
+{signal: \[ {node: '.a..b........', phase:0.2},
+
+{name: 'adc\_pd\_i' , wave: '10|..|.....|....|..1'},
+
+{name: 'clk\_ast\_adc\_i', wave: 'p.|..|.....|....|...'},
+
+{name: 'adc\_chnsel\_i' , wave: '0.|.3|..04.|....|0..'},
+
+{name: 'adc\_d\_val\_o' , wave: '0.|..|.1.0.|.1..|.0.'},
+
+{name: 'adc\_d\_o' , wave: 'x.|..|.3.x.|.4..|.x.', data: \['ch0', 'ch1',
+'ch1'\]},
+
+\],edge: \[ 'a\<-\>b wakeup time', \] }
+
+# Random Number Generator
+
+AST contains a random number generator that outputs random number
+bitstreams whenever it is enabled. After enabled by the [<u>comportable
+controller</u>](https://github.com/lowRISC/opentitan/blob/master/hw/ip/entropy_src/doc/_index.md)
+through ‘rng\_en\_i’, the AST begins generating multiple independent
+four random bit streams when \`rng\_ok\_o\` is asserted.
+
+The comportable controller then samples the rng bitstream at a
+configurable rate (random data is produced at \~50KHz rate). It is
+important to note that rng\_ok is NOT used to indicate when to sample,
+but simply that the rng is in a valid operational state. It is up to the
+comportable controller to sample at the correct rate to ensure a
+particular random bit is not consumed more than once. This rate will be
+known in advance and configured by software. For more information on how
+the OpenTitan entropy system operates, please see the most recent.
+
+# Entropy Consumption
+
+AST consumes entropy for defensive purposes. However, AST does not
+consume its raw entropy directly. Instead, AST receives entropy from the
+[<u>CSRNG</u>](https://github.com/lowRISC/opentitan/tree/master/hw/ip/csrng).
+The details of this interface are still under discussion.
+
+# Countermeasures and Alerts
+
+## Alert Events
+
+AST’s sensors and detectors, when triggered, output alert events to a
+sensor controller. The event signals are level until acknowledged by the
+controller. Further, the events are differentially encoded to ensure
+they cannot be hard-wired or faulted to either ‘1’ or ‘0’.
+
+Inside the sensor controller, the events are then converted into alerts
+as part of the wider [<u>OpenTitan alert handling
+system</u>](https://github.com/lowRISC/opentitan/blob/master/hw/ip/alert_handler/doc/_index.md).
+
+## Alert Signaling
+
+Outgoing alert events are level. Incoming event ack signals clear the
+alert event (similar to an interrupt). Outgoing alert events should be
+OR'd inside the sensor or power manager (depending on what level of deep
+sleep support is needed) to generate wakeup, that way AST does not need
+to do any additional handling for wakeups during low power mode.
+
+## Countermeasures
+
+Most countermeasure enablement is controlled by Nuvoton via the

```

Comment:
This sentence is a little confusing to me.  Should it say "Most countermeasures are enabled via the AST register interface." ?

If that's correct, then is clock jitter controlled by something other than the register interface?  (Or is there logic inside AST doing this dynamic enable/disable, on top of some register-based request?)

Reply:
hey chris, the jittery feature specifically has an hardware enable port coming from the outside world. 
So the clock controller on opentitan side would actually maintain the register interface.

Reply:
Ohhhh.  I see.  All fine then.